plugin/modify-env/modify-env.c

```diff
@@ -48,7 +49,16 @@ void dmtcp_event_hook(DmtcpEvent_t event, DmtcpEventData_t *data)
   switch (event) {
   case DMTCP_EVENT_RESTART:
   { int size = 12288;
-    char *buf = read_dmtcp_env_file("dmtcp_env.txt", size);
+    int retval;
+    char env_file[PATH_MAX];
+
+    retval = dmtcp_get_restart_env("DMTCP_ENV_FILE", env_file, PATH_MAX);
+    if (retval != 0)
+    {
+        strcpy(env_file, "dmtcp_env.txt"); 
```

Comment:
While we're exchanging possible styles, here's another posibility.  This doesn't need macros.

```
       strncpy(env_file, "dmtcp_env.txt", sizeof "dmtcp_env.txt"); 
```
The sizeof operator will also account for the null char at the end of the string.

I also agree with @kwharrigan that the original `strcpy` is probably good enough here.  The declaration of env_file uses `PATH_MAX`, which is surely larger than `sizeof "dmtcp_env.txt"`.